dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/config/HibernateEncryptionConfig.java

```diff
@@ -101,7 +101,9 @@ public MethodInvokingFactoryBean addProvider() {
   }
 
   private Object getConnectionProperty(String key, String defaultValue) {
-    String value = hibernateConfigurationProvider.getConfiguration().getProperty(key);
+    String value =
+        dhisConfigurationProvider.getPropertyOrDefault(
+            ConfigurationKey.ENCRYPTION_PASSWORD, "J7GhAs287hsSQlKd9g5");
```

Comment:
this key is marked as sensitive, should we be explicitly setting this in plain sight here or should we be doing something else?
I'm not sure how this key is used, if it's not set in the config file, does it cause issues if it's not set?

Reply:
This is suboptimal and is well known.
We should not use a default password here but rather generate a unique on first boot, outside this process and inject it safely. The facilities to do this is lacking and requires a bigger refactoring and feature work to get in place. It is on the security backlog and roadmap.

Reply:
great